lib/src/convert_es6_proxy.html

```
@@ -0,0 +1,120 @@
+    };
+
+    ES6.createES6JsProxy = function(instance) {
+      return new Proxy({},{
+        'get' : function(target,propertyName,proxy) {
+
+          if (commonBehaviors.hasOwnProperty(propertyName)) {
+            return commonBehaviors[propertyName](instance);
+          }
+
+          var res = ES6._dartGetter(instance,propertyName);
+          if(res === ES6.Unsupported) {
+            return target[propertyName];
+          } else {
+            return res;
+          }
+        },
+        'set' : function(target,propertyName,value,proxy) {
+          if (ES6._dartSetter(instance,propertyName,value) === ES6.Unsupported) {
+            target[propertyName] = value;
+          }
+          return true;
+        }
+      });
+    };
+
+    /**
+     * List ES6JsProxy support
+     */
+
+    var arrayCommonBehaviors = {
+      '__dartClass__' : function(instance) { return instance; },
+      '__isES6Proxy__' : function(instance) { return true; },
+      'splice' : function (instance) {
+        return function(index,howmany) {
+          return ES6._dartArraySplice(instance,index,howmany,Array.prototype.slice.call(arguments,2));
+        };
+      },
+      'slice' : function(instance) {
+        return function(begin,end) {
+          return ES6._dartArraySlice(instance,begin,end);
+        };
+      },
+      'length' : function(instance) { return ES6._dartArrayLength(instance); },
+      'push' : function(instance) {
+        return function() {
+          return ES6._dartArrayPush(instance,Array.prototype.slice.call(arguments));
+        };
+      }
+    };
+
+    ES6.createES6JsProxyForArray = function(instance) {
+      return new Proxy([],{
+        'get' : function(target,propertyName,proxy) {
+
+          if (arrayCommonBehaviors.hasOwnProperty(propertyName)) {
+            return arrayCommonBehaviors[propertyName](instance);
+          }
+
+          if (isNaN(propertyName)) {
+            return target[propertyName];
```

Same as above, I would throw or return undefined.